source/wp-content/themes/wporg-developer-2023/src/version-select/index.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Block Name: Version Select
+ * Description: Displays WordPress versions in a select element.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\Theme\Developer_2023\Dynamic_Version_Select;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/version-select',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+}
+
+/**
+ * Render the block content.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+	if ( ! isset( $block->context['postId'] ) ) {
+		return '';
+	}
```

Comment:
I don't think this block needs a post context, does it?

Reply:
Nope.